File: src/main/java/net/moltendorf/Bukkit/SkyrimAppetite/SkyrimAppetite.java
package net.moltendorf.Bukkit.SkyrimAppetite;

import org.bukkit.plugin.java.JavaPlugin;

/**
 * Created by moltendorf on 2016-02-28.
 */
public class SkyrimAppetite extends JavaPlugin {

	// Main instance.
	private static SkyrimAppetite instance = null;

	protected static SkyrimAppetite getInstance() {
		return instance;
	}

	// Variable data.
	protected Listeners listeners = null;
	protected Settings settings = null;

	@Override
	public void onEnable() {
		instance = this;

		// Construct new settings.
		listeners = new Listeners();
		settings = new Settings();

		getServer().getPluginManager().registerEvents(listeners, this);
	}

	@Override
	public void onDisable() {
		settings.save();

		instance = null;
	}
}

File: src/main/kotlin/net/moltendorf/bukkit/skyrimappetite/SkyrimAppetite.kt
package net.moltendorf.bukkit.skyrimappetite

import net.moltendorf.bukkit.skyrimappetite.settings.GlobalSettings
import org.bukkit.plugin.java.JavaPlugin

/**
 * Created by moltendorf on 2016-02-28.
 *
 * @author moltendorf
 */
class SkyrimAppetite : JavaPlugin() {
  // Variable data.
  lateinit var listeners: Listeners
    private set
  lateinit var settings: GlobalSettings
    private set

  override fun onEnable() {
    instance = this

    // Construct new settings.
    settings = GlobalSettings()

    // Are we enabled?
    enabled = settings.enabled

    if (enabled) {
      // Register listeners.
      listeners = Listeners()
      server.pluginManager.registerEvents(listeners, this)
      log.info("Enabled general listeners.")
    }
  }

  override fun onDisable() {
    settings.save()

    enabled = false
  }

  companion object {
    var enabled = false
      private set

    lateinit var instance: SkyrimAppetite
      private set
  }
}
